js/web/lib/wasm/jsep/webgpu/ops/binary-op.ts

```diff
@@ -64,13 +66,15 @@ const createBinaryOpProgramShader =
       if (vectorize) {
         if (doBroadcast) {
           assignment = `
-      ${outputIndicesHelper.indicesVariableDeclaration('outputIndices')}
-      ${outputIndicesHelper.o2iCall('global_idx * 4u', 'outputIndices')}
-      let offsetA = calcOffsetA(&outputIndices);
-      let offsetB = calcOffsetB(&outputIndices);
-      outputData[global_idx] = ${expressionVector('aData[offsetA / 4u]', 'bData[offsetB / 4u]')};`;
+      let outputIndices = ${output.offsetToIndices('global_idx * 4u')};
+      let offsetA = calcOffsetA(outputIndices);
+      let offsetB = calcOffsetB(outputIndices);
+      ${
+              output.setByOffset(
+                  'global_idx', expressionVector(a.getByOffset('offsetA / 4u'), b.getByOffset('offsetB / 4u')))}`;
```

Comment:
binary-op.ts is not well optimized in this version and because it's written before indices helper is introduced, some source code is hardcoded to use component of '4' in this source code, so this is why we use something like `offset / 4`.

IndicesHelper does not use this kind of offset specification in normal usage.